lib/client.js

```diff
@@ -400,6 +403,15 @@ MqttClient.prototype.publish = function (topic, message, opts, callback) {
   return this
 }
 
+MqttClient.prototype._checkSubscribedTopic = function (topic) {
+  if (!~~this._subscribedTopics[topic]) {
+    this._subscribedTopics[topic] = true
```

Comment:
this should be QoS aware, multiple subscribes with different QoS should be treated separately.